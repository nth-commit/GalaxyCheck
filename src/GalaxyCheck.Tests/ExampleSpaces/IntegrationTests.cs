using Snapshooter;
using Snapshooter.Xunit;
using Xunit;
using GC = GalaxyCheck;
using GalaxyCheck.ExampleSpaces;

namespace Tests.ExampleSpaces
{
    public class IntegrationTests
    {
        [Theory]
        [InlineData(1, 0)]
        [InlineData(2, 0)]
        [InlineData(5, 0)]
        [InlineData(10, 0)]
        public void IntegerExampleSpace(int value, int target)
        {
            var exampleSpace = ExampleSpaceFactory.Unfold(
                value,
                ShrinkFunc.Towards(target),
                x => x,
                IdentifyFuncs.Default<int>());

            Snapshot.Match(exampleSpace.Render(x => x.ToString()), SnapshotNameExtension.Create(value, target));
        }

        [Fact]
        public void IntegerExampleSpaceFiltered()
        {
            var exampleSpace = ExampleSpaceFactory
                .Unfold(
                    10,
                    ShrinkFunc.Towards(0),
                    x => x,
                    IdentifyFuncs.Default<int>())
                .Filter(x => x % 2 == 0);

            Snapshot.Match(exampleSpace?.Render(x => x.ToString()) ?? "");
        }
    }
}
