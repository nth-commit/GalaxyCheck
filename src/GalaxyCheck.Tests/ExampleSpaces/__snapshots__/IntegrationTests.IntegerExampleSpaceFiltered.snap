10
..0
..8
....4
......2
....6
